Source/WebCore/page/Performance.cpp

```diff
@@ -321,6 +321,12 @@ void Performance::processEventEntry(const PerformanceEventTimingCandidate& candi
     if (!m_firstInput && candidate.interactionID.value) {
         m_firstInput = PerformanceEventTiming::create(candidate, true);
         queueEntry(*m_firstInput);
+        if (is<Document>(scriptExecutionContext())) {
```

Comment:
I think there's a better place for this code. Setting `window->setDispatchedInputEvent()` should happen somewhere in the call stack, before we call down into `Performance`, because this state isn't specifically about performance-related stuff. Maybe somewhere in EventDispatcher?